src/lib.rs

```diff
@@ -248,9 +277,63 @@ impl Bloom {
     #[classattr]
     const __hash__: Option<PyObject> = None;
 
-    /// Load from a file, see "Persistence" section in the README
+    /// Load from a file, fileobj or bytes, see "Persistence" section in the README
     #[classmethod]
-    fn load(_cls: &PyType, filepath: &str, hash_func: &PyAny) -> PyResult<Bloom> {
+    fn load(_cls: &PyType, src: PyObject, hash_func: &PyAny) -> PyResult<Bloom> {
+        match SaveLoadType::from_pyobject(Some(src)) {
+            Ok(f) => match f {
+                SaveLoadType::Filepath(filepath) => {
+                    let file = File::open(filepath)?;
+                    Self::load_from_read(hash_func, || Ok::<_, PyErr>(file))
+                }
+                SaveLoadType::FileLike(f) => Self::load_from_read(hash_func, || Ok::<_, PyErr>(f)),
+                SaveLoadType::Bytes(b) => Self::load_from_read(hash_func, || {
+                    Ok::<&[u8], std::convert::Infallible>(b.as_ref())
+                }),
+                _ => Err(PyErr::new::<PyTypeError, _>("Incorrect argument type")),
+            },
+            Err(e) => Err(e),
+        }
+    }
```

Comment:
If we implement the other suggestions, this can be just:

```rust
    fn load(_cls: &PyType, src: LoadSrc, hash_func: &PyAny) -> PyResult<Bloom> {
        match src {
            LoadSrc::Filepath(filepath) => Self::load_from_read(hash_func, || File::open(filepath)),
            LoadSrc::FileLike(f) => Self::load_from_read_obj(hash_func, f),
            LoadSrc::Bytes(b) => Self::load_from_read_obj(hash_func, b),
        }
    }
```